lib/types/btrfs.nix

```diff
@@ -126,8 +126,10 @@ in
             MNTPOINT=$(mktemp -d)
             mount ${config.device} "$MNTPOINT" -o subvol=/
             trap 'umount $MNTPOINT; rm -rf $MNTPOINT' EXIT
-            btrfs subvolume create "$MNTPOINT"/${subvol.name} ${toString subvol.extraArgs}
-            ${swapCreate "$MNTPOINT/${subvol.name}" subvol.swap}
+            SUBVOL_ABS_PATH="$MNTPOINT/${subvol.name}"
+            mkdir -p "''${SUBVOL_ABS_PATH%/*}"
```

Comment:
usually we try to avoid bash features, since people also use disko on non bash platforms (like openwrt)

Reply:
Thanks for your review! Afaik, this particular parameter expansion is [required by POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02). Based on my tests, recent versions of BusyBox's `ash` (used by OpenWrt) can also handle it.

 I'm happy to modify this to accommodate edge cases but would need to know what to target. We could replace it w/command substitution (e.g., using `dirname`, which is also [required by POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/dirname.html), or `awk`, `grep`, `sed`, ...), but given the choice, I tend to prefer parameter expansion to prevent invoking a separate process.

Reply:
...or the third option: Nix! 🤭

I've updated the PR to use the `lib.dirOf` function.

Reply:
I think we also should at what default system shells actually supports rather than what some out-dated standard says. Ash has a lot more features than your grandfathers sh.